### PR TITLE
Feature/processing data in models

### PR DESCRIPTION
**Pull request by Salimer**
### Create models
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
- Set up associations between models.
    - Remember that author_id column in posts table should be the foreign_key for the users table.

### Use models to insert data

- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
```
   first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
```

- Create at least 4 posts written by one of the users you created by running the following code:

```
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
```

- Create at least 6 posts comments for one of the posts you created by running the following code:

```
Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
Your models should include some additional methods.
- Users
    - A method that returns the 3 most recent posts for a given user.
- Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
- Comments
    - A method that updates the comments counter for a post.
- Likes
    - A method that updates the likes counter for a post.
- Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  # Attributes
+  attribute :name, :string
+  attribute :bio, :text
+  attribute :posts_counter, :integer, default: 0
+  attribute :photo, :string
+
+  # Methods
+  def self.three_most_recent_posts(user)
+    user.posts.order(created_at: :desc).limit(3)
```
Comment (fobadara) [inline app/models/user.rb]: [OPTIONAL] There was no need to get pass user as a parameter. Rails knows that `posts.order..` means `User.posts.order...`.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  # Attributes
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_counter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  # Methods
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
```
Comment (fobadara) [inline app/models/post.rb:18]: [OPTIONAL] It is a good practice to use rails callbacks like `after_save` in scenarios like this to call the custom method after creation.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  # Associations
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  # Methods
+  def update_post_likes_counter
```
Comment (fobadara) [inline app/models/like.rb:10]: [OPTIONAL] It is a good practice to use rails callbacks like `after_save` in scenarios like this to call the custom method after creation.

**Review by fobadara — APPROVED** [review]
Hi @Salimer, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good use of separation of concerns


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Salimer) [reply app/models/user.rb]: Addressed

File: app/models/post.rb (hunk 2 above)
Comment (Salimer) [reply app/models/post.rb:18]: Addressed

File: app/models/like.rb (hunk 3 above)
Comment (Salimer) [reply app/models/like.rb:10]: Addressed

**Comment by Salimer** [comment]
Thank you @fobadara for the thorough review, all optional changes are addressed 🥳

**Pull request merged**